core/server/auth/oauth.js

```diff
@@ -62,6 +63,44 @@ function exchangePassword(client, username, password, scope, done) {
         });
 }
 
+function exchangeAuthorizationCode(req, res, next) {
+    if (!req.body.authorizationCode) {
+        return next(new errors.UnauthorizedError({
+            message: i18n.t('errors.middleware.auth.accessDenied')
+        }));
+    }
+
+    req.query.code = req.body.authorizationCode;
+
+    passport.authenticate('ghost', {session: false, failWithError: false}, function authenticate(err, user) {
+        if (err) {
+            return next(new errors.UnauthorizedError({
+                err: err
+            }));
+        }
+
+        if (!user) {
+            return next(new errors.UnauthorizedError({
+                message: i18n.t('errors.middleware.auth.accessDenied')
+            }));
+        }
+
+        authenticationAPI.createTokens({}, {context: {client_id: req.client.id, user: user.id}})
+            .then(function then(response) {
+                spamPrevention.resetCounter(user.get('email'));
```

Comment:
Authorization codes are exempt from spam prevention according to the change here: https://github.com/TryGhost/Ghost/pull/7571/commits/28d40946234082c029fd77496d07590e07ffed6c#diff-7b513b6d183ddea58102b60e32c8b251R30. So no need to reset the counter :).